master/buildbot/test/unit/test_fake_secrets_manager.py

```diff
@@ -0,0 +1,88 @@
+class TestSecretsManager(unittest.TestCase):
+
+    def setUp(self):
+        self.master = fakemaster.make_master()
+        self.master.config.secretsManagers = [FakeSecretStorage({"foo": "bar",
+                                                                 "other": "value"})]
+
+    @defer.inlineCallbacks
+    def testGetManagerService(self):
+        secret_service_manager = SecretManager()
+        SecretManager.master = self.master
+        expectedClassName = FakeSecretStorage.__name__
+        expectedSecretDetail = SecretDetails(expectedClassName, "foo", "bar")
+        secret_result = yield secret_service_manager.get("foo")
+        strExpectedSecretDetail = str(secret_result)
+        self.assertEqual(secret_result, expectedSecretDetail)
+        self.assertEqual(secret_result.key, "foo")
+        self.assertEqual(secret_result.value, "bar")
+        self.assertEqual(secret_result.source, expectedClassName)
+        self.assertEqual(strExpectedSecretDetail,
+                         "FakeSecretStorage foo: 'bar'")
+
+    @defer.inlineCallbacks
+    def testGetNoDataManagerService(self):
+        secret_service_manager = SecretManager()
+        SecretManager.master = self.master
+        secret_result = yield secret_service_manager.get("foo2")
+        self.assertEqual(secret_result, None)
+
+    @defer.inlineCallbacks
+    def testGetDataMultipleManagerService(self):
+        secret_service_manager = SecretManager()
+        self.master.config.secretsManagers = [FakeSecretStorage({"foo": "bar",
+                                                                 "other": "value"}),
+                                              OtherFakeSecretStorage({"foo2": "bar",
+                                                                      "other2": "value"},
+                                                                     props={"property": "value_prop"})
+                                              ]
+        SecretManager.master = self.master
+        expectedSecretDetail = SecretDetails(OtherFakeSecretStorage.__name__,
+                                             "foo2",
+                                             "bar")
+        secret_result = yield secret_service_manager.get(
+            "foo2")
+        self.assertEqual(secret_result, expectedSecretDetail)
+
+    @defer.inlineCallbacks
+    def testGetDataMultipleManagerServiceNoDatas(self):
+        secret_service_manager = SecretManager()
+        self.master.config.secretsManagers = [FakeSecretStorage({"foo": "bar",
+                                                                 "other": "value"}),
+                                              FakeSecretStorage({"foo2": "bar",
+                                                                 "other2": "value"})
+                                              ]
+        SecretManager.master = self.master
+        secret_result = yield secret_service_manager.get("foo3")
+        self.assertEqual(secret_result, None)
```

Comment:
I don't see the test with two providers providing the same secret.

Need another test with a secret value being empty string ( empty string is valid secret)